spec.html

```diff
@@ -27821,28 +27821,29 @@ <h1>Math.floor ( _x_ )</h1>
         <h1>Math.fround ( _x_ )</h1>
         <p>When the `Math.fround` method is called with argument _x_, the following steps are taken:</p>
         <emu-alg>
-          1. If _x_ is *NaN*, return *NaN*.
-          1. If _x_ is one of *+0*, *-0*, *+&infin;*, *-&infin;*, return _x_.
-          1. Let _x32_ be the result of converting _x_ to a value in IEEE 754-2019 binary32 format using roundTiesToEven mode.
+          1. Let _n_ be ? ToNumber(_x_).
+          1. If _n_ is *NaN*, return *NaN*.
+          1. If _n_ is one of *+0*, *-0*, *+&infin;*, *-&infin;*, return _n_.
+          1. Let _x32_ be the result of converting _n_ to a value in IEEE 754-2019 binary32 format using roundTiesToEven mode.
           1. Let _x64_ be the result of converting _x32_ to a value in IEEE 754-2019 binary64 format.
           1. Return the ECMAScript Number value corresponding to _x64_.
         </emu-alg>
       </emu-clause>
 
       <emu-clause id="sec-math.hypot">
-        <h1>Math.hypot ( _value1_, _value2_, ..._values_ )</h1>
+        <h1>Math.hypot ( ..._values_ )</h1>
         <p>Returns the square root of the sum of squares of its arguments.</p>
-        <p>When the `Math.hypot` method is called with at least two arguments _value1_ and _value2_ and any number of additional arguments which form the rest parameter ..._values_, the following steps are taken:</p>
+        <p>When the `Math.hypot` method is called with zero or more arguments which form the rest parameter ..._values_, the following steps are taken:</p>
         <emu-alg>
-          1. Let _numbers_ be a List containing _value1_, _value2_, and the elements of _values_ in List order.
           1. Let _onlyZero_ be *true*.
-          1. For each element _number_ of _numbers_, do
-            1. Let _n_ be ? ToNumber(_number_).
+          1. For each element _value_ of _values_, do
+            1. Let _n_ be ? ToNumber(_value_).
             1. If _n_ is *NaN*, _n_ is *+&infin;*, or _n_ is *-&infin;*, return _n_.
             1. If _n_ is neither *+0* nor *-0*, set _onlyZero_ to *false*.
           1. If _onlyZero_ is *true*, return *+0*.
-          1. Return an implementation-approximated value representing the square root of the sum of squares of the elements of _numbers_.
+          1. Return an implementation-approximated value representing the square root of the sum of squares of the elements of _values_.
         </emu-alg>
+        <p>The *"length"* property of the `hypot` method is 2.</p>
```

Comment:
i'm not convinced on this approach; i strongly prefer to minimize the occurrence of explicit length notes.

Reply:
The only way to eliminate these three explicit length statements would be for each function to have a heading with exactly 2 "normal" parameters, which would be mislead (or confuse) the reader: it would suggest that:
- a 'proper' call to the function requires at least 2 arguments, or
- if a call supplies less than 2 args, the 'missing' args would be `*undefined*`.

But neither is correct for these functions.

Personally, I think it's more important to clearly convey the function's semantics. If that results in the need for an explicit length statement, that doesn't bother me.

In fact, I think it's *better* to have an explicit length statement in these cases, because it underlines that the length doesn't 'match' the semantics.

Reply:
@ljharb I entirely agree with @jmdyck's reasoning here.

Reply:
@jmdyck's reasoning is compelling here.

If nothing else, the current rendering is very confusing because "required parameter" within the spec has a very narrow definition, which is "rendered not as a rest parameter and not between [ ]". It so happens that that narrow definition mostly aligns with the broader and more common understanding of "required" that means "needed for the function to do the useful thing". The functions for which these definitions don't align are these Math functions, and so the explicit "length" notes are definitely editorially warranted IMO.